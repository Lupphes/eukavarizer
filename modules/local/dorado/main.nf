process DORADO {
    tag "$meta.id"
    label 'process_medium'

    conda "${moduleDir}/environment.yml"
    container "docker.io/nanoporetech/dorado:sha268dcb4cd02093e75cdc58821f8b93719c4255ed"

    input:
    tuple val(meta), path(fast5s)

    output:
    tuple val(meta), path("*.fastq.gz"), emit: fastq
    path "versions.yml"                , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def model = task.ext.model ?: 'dna_r9.4.1_e8_hac@v3.3'
    """
    dorado basecaller \\
        ${model} \\
        ${args} \\
        ${fast5s} \\
        | gzip > ${prefix}.fastq.gz

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        dorado: \$( dorado --version 2>&1 | sed 's/dorado //g' )
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    printf "@read1\\nACGTACGT\\n+\\n!!!!!!!!\\n" | gzip > ${prefix}.fastq.gz

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        dorado: \$( dorado --version 2>&1 | sed 's/dorado //g' )
    END_VERSIONS
    """
}
