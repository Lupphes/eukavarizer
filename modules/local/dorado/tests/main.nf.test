nextflow_process {

    name "Test Process DORADO"
    script "../main.nf"
    process "DORADO"
    tag "dorado"
    tag "basecalling"
    tag "modules"
    tag "modules_local"
    tag "nanopore"

    test("Basecall nanopore reads - stub") {

        options '-stub'

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                // Note: Using stub mode since fast5 files are large and not in test data
                input[0] = channel.of([
                    [ id: 'nanopore_test' ],
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz')
                ])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() },
                { assert process.out.fastq },
                { assert process.out.versions }
            )
        }
    }

    test("Basecall with custom model config - stub") {

        options '-stub'
        config './nextflow.config'

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = channel.of([
                    [ id: 'custom_model' ],
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz')
                ])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() },
                { assert process.out.fastq },
                { assert process.out.fastq.get(0).get(1).toString().endsWith('.fastq.gz') },
                { assert process.out.versions }
            )
        }
    }
}
