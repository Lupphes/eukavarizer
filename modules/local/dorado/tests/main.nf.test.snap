{
    "Basecall with custom model config - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "custom_model"
                        },
                        "custom_model.fastq.gz:md5,8773e2bde287287c737f0e4b1351c05c"
                    ]
                ],
                "1": [
                    "versions.yml:md5,f8f2bece6ba0760ba4c144d3d7ad468d"
                ],
                "fastq": [
                    [
                        {
                            "id": "custom_model"
                        },
                        "custom_model.fastq.gz:md5,8773e2bde287287c737f0e4b1351c05c"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,f8f2bece6ba0760ba4c144d3d7ad468d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.04.8"
        },
        "timestamp": "2025-10-20T22:35:25.206315425"
    },
    "Basecall nanopore reads - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "nanopore_test"
                        },
                        "nanopore_test.fastq.gz:md5,8773e2bde287287c737f0e4b1351c05c"
                    ]
                ],
                "1": [
                    "versions.yml:md5,f8f2bece6ba0760ba4c144d3d7ad468d"
                ],
                "fastq": [
                    [
                        {
                            "id": "nanopore_test"
                        },
                        "nanopore_test.fastq.gz:md5,8773e2bde287287c737f0e4b1351c05c"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,f8f2bece6ba0760ba4c144d3d7ad468d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.04.8"
        },
        "timestamp": "2025-10-20T22:35:17.713014003"
    }
}